ocf_data_sampler/sample/base.py

```diff
@@ -0,0 +1,57 @@
+# base.py
+
+"""
+Base class definition - abstract
+Handling of both flat and nested structures - consideration for NWP
+"""
+
+import logging
+from pathlib import Path
+from typing import Any, Dict, Optional, Union
+from abc import ABC, abstractmethod
+
+logger = logging.getLogger(__name__)
+
+class SampleBase(ABC):
+    """ 
+    Abstract base class for all sample types 
+    Provides core data storage functionality
+    """
+
+    def __init__(self):
```

Comment:
, should we add `data` to the _init_ so we can set `_data` when we create Sample straight away